app/src/main/kotlin/compiler/intermediate/generators/GeneratorDetailsGenerator.kt

```diff
@@ -0,0 +1,29 @@
+package compiler.intermediate.generators
+
+import compiler.ast.Statement
+import compiler.intermediate.ControlFlowGraph
+import compiler.intermediate.IFTNode
+
+interface GeneratorDetailsGenerator : VariableAccessGenerator {
```

Comment:
I'm not a huge fan of this name, but we can leave it as some kind of a joke, or change it to `GeneratorDetailsCreator`.

Reply:
Actually, I really like that name (also, it is consistent with `FunctionDetailsGenerator`), so I'll keep it.